railties/lib/rails/generators/rails/app/templates/rubocop.tt

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env ruby
```

Comment:
I'm not sure this is what we had in mind, I just generated the file using `bundle binstubs`, but we can just run that command in the generator.

Reply:
You can checkout https://github.com/rails/rails/commit/25c649c5132dcd2c979a8f2b2392e7898872d144 as an alternative way to copy over the binstub. I also did not copy over the complete generated binstub to be more in line with the existing ones.

Also I also edited some more generated files to align with the default configuration, basically I tried to keep the generated app green even for commented out code.

Also at least on my setup, running Rubocop in an actual new app did not work without https://github.com/rails/rubocop-rails-omakase/pull/4. Can you reproduce this as well or did I miss sth. here?

Reply:
I fixed the rubocop-packaging thing in 8f81b8c

Reply:
Ah, so I did miss sth.

I just checked out your branch and generated a new app and noticed the following small things:

* The generated binstub does not have execute permissions
* Running `bundle exec rubocop` does still fail with the packaging/md issue. Maybe it's okay to just add them to the omakase gem and avoid those issues :)

Thank your for your work on the PR 👍

Reply:
> The generated binstub does not have execute permissions

Good catch, I was comparing the permissions _before_ generation and fixed that in 894cdec.

> Running bundle exec rubocop does still fail with the packaging/md issue. Maybe it's okay to just add them to the omakase gem and avoid those issues :)

Weird, I thought that commit fixed it, but I am also perturbed that the workflow doesn't fail when `rails new` errors due to that. The reason is because rubocop looks for `.rubocop.yml` files recursively upwards (afaict) and discovers the one in the parent directory (a checkout of rails/rails), which depends on `rubocop-packaging` and `rubocop-md`. That is not what we want, and the point of that `-c` flag was supposed to circumvent.

Reply:
Ok, I just was calling `rubocop` and not the `bin/rubocop` when generating the application, should be fixed now in a7f56d1a1a.

Reply:
And the `run "bin/rubocop"` command from Thor doesn't _fail_ in this case because we aren't using `abort_on_failure` option, so this is expected behavior.

Probably makes sense to not raise the red flag if rubocop fails when generating a new application? That doesn't help escalate changes which might introduce rubocop errors to the default setup though.

Reply:
You're right, it was broken because I generated an app in my Rails checkout, thx for your explanation. I'll close my PR and link this thread for an explanation :)
Binstub works fine now, I just checked 👍